sklearn/tests/test_common.py

```diff
@@ -297,3 +304,186 @@ def test_search_cv(estimator, check, request):
 def test_check_n_features_in_after_fitting(estimator):
     _set_checking_parameters(estimator)
     check_n_features_in_after_fitting(estimator.__class__.__name__, estimator)
+
+
+class MinimalClassifier:
+    """Minimal classifier implementation with inheriting from BaseEstimator."""
+    _estimator_type = "classifier"
+
+    def __init__(self, param=None):
+        self.param = param
+
+    def __repr__(self):
+        # Only required when using pytest-xdist to get an id not associated
+        # with the memory location
+        return self.__class__.__name__
+
+    def get_params(self, deep=True):
+        return {"param": self.param}
+
+    def set_params(self, **params):
+        for key, value in params.items():
+            setattr(self, key, value)
+        return self
+
+    def __getstate__(self):
+        return self.__dict__.copy()
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
```

Comment:
I don't understand why we would require third party estimators to implement those methods. We just ask them to be picklable which is a the case by default for most Python classes. No need to impose this extra boiler plate.

Reply:
Uhm I probably got confused with a pickle failure which was due to the version. Since we change the common check, this is not necessary indeed.